### PR TITLE
Ensuring ipv4 config ahead of ipv6 in /etc/network/interfaces file for mgmt interface

### PR DESCRIPTION
**Pull request by wendani**
In generating the interfaces file, put ipv4 config ahead of ipv6 for the mgmt interface

This solves the problem that the mgmt interface does not get its ipv4 address

Signed-off-by: Wenda <wenni@microsoft.com>

**- What I did**
Put ipv4 config ahead of ipv6 for the mgmt interface. I observe that on one DUT, the mgmt interface does not possess its IPv4 address and fails to be brought up if the other way round.

**- How I did it**
Change interfaces.j2

**- How to verify it**
reboot and see if the mgmt interface has IPv4 address and is reachable.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: files/image_config/interfaces/interfaces.j2 [hunk 1]
```diff
@@ -21,7 +21,7 @@ iface lo {{ 'inet' if prefix | ipv4 else 'inet6' }} static
 # The management network interface
 auto eth0
 {% if MGMT_INTERFACE %}
-{% for (name, prefix) in MGMT_INTERFACE %}
+{% for (name, prefix) in MGMT_INTERFACE|sort %}
```
Comment (lguohan) [inline files/image_config/interfaces/interfaces.j2]: why sort put v4 ahead of v6? is that a co-incidence?

Comment (wendani) [reply files/image_config/interfaces/interfaces.j2]: The name field is the same, eth0, for both v4 and v6. The sort is based on what is taken in the prefix

Comment (lguohan) [reply files/image_config/interfaces/interfaces.j2]: there is no guarantee that v4 prefix will be ahead of v6 prefix.

File: src/sonic-config-engine/sonic-cfggen [hunk 2]
```diff
@@ -33,6 +33,11 @@ from swsssdk import ConfigDBConnector
 from collections import OrderedDict
 from natsort import natsorted
 
+def sift_up_ipv4(value):
```
Comment (lguohan) [inline src/sonic-config-engine/sonic-cfggen]: sort_by_prefix

add comments:
for interface name is the same, then sort by prefix. v4 address is before v6 address

**Comment by lguohan** [comment]
Need unit test on interfaces.j2 file.

**Comment by wendani** [comment]
@stcheng src/sonic-config-engine/tests/sample_output has the correct order--v4 profile ahead of v6

**Comment by lguohan** [comment]
root caused in #1701. drop this PR.

**Pull request closed without merging**